src/components/popover-container/popover-container.spec.tsx

```diff
@@ -785,6 +808,119 @@ describe("open state when click event triggered", () => {
     expect(onCloseFn).not.toHaveBeenCalled();
   });
 
+  it("should close the container and call onClose when controlled and escape key is pressed", () => {
+    const onCloseFn = jest.fn();
+    const MockWrapper = () => {
+      const [open, setOpen] = React.useState(true);
+
+      return (
+        <PopoverContainer
+          title="PopoverContainerSettings"
+          open={open}
+          onClose={(e) => {
+            setOpen(false);
+            onCloseFn(e);
+          }}
+        />
+      );
+    };
+    mount(<MockWrapper />);
+
+    document.dispatchEvent(
+      new KeyboardEvent("keydown", {
+        key: "Escape",
+        bubbles: true,
+      })
+    );
+    expect(onCloseFn).toHaveBeenCalled();
+  });
+
+  it("should close the container when escape key is pressed inside of a Select component", () => {
+    const onCloseFn = jest.fn();
+    const MockWrapper = () => {
+      const [open, setOpen] = React.useState(true);
+
+      return (
+        <>
+          <PopoverContainer
+            title="PopoverContainerSettings"
+            open={open}
+            onClose={(e) => {
+              setOpen(false);
+              onCloseFn(e);
+            }}
+          >
+            <Select name="simple" id="simple" label="color" labelInline>
+              <Option text="Amber" value="1" />
+              <Option text="Black" value="2" />
+            </Select>
+          </PopoverContainer>
+        </>
+      );
+    };
+    const wrapper = mount(<MockWrapper />);
+    expect(wrapper.update().find(PopoverContainer).prop("open")).toBe(true);
+
+    const selectInput = document.querySelector(
+      '[data-element="input"][aria-expanded="false"]'
+    );
+
+    selectInput?.dispatchEvent(
+      new KeyboardEvent("keydown", {
+        key: "Escape",
+        bubbles: true,
+      })
+    );
+
+    expect(wrapper.update().find(PopoverContainer).exists()).toBe(true);
```

Comment:
**question**: even the test description, is this assertion meant to be checking the `PopoverContainer` has been closed?

```suggestion
    expect(wrapper.update().find(PopoverContainer).exists()).toBe(false);
```

Reply:
Same again here. I'll check that the content does not exist.